clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -537,51 +586,137 @@ private void updateGroupSubscription(Set<String> topics) {
         return groupAssignment;
     }
 
+    /**
+     * Used by COOPERATIVE rebalance protocol only.
+     *
+     * Validate the user assignor returned assignments such that there are no partitions which are going to
+     * be assigned to a different consumer other than its current owner (i.e. who includes it in its owned
+     * partitions list encoded in subscription); also if there are any revoked partitions set the error code to let
+     * everyone retry.
+     */
+    private void validateCooperativeAssignment(final Map<String, List<TopicPartition>> ownedPartitions,
```

Comment:
Just making sure I understand this. I think the main thing we're trying to validate is that for each member, the new assignment contains all the owned partitions. is there additional validation needed?

Reply:
I think the idea is to validate that the new assignment doesn't contain any partitions that are being revoked during this rebalance

Reply:
@ableegoldman is right here: we just want to make sure no prev-owned-partitions have been re-assigned.